snap/gadget.go

```diff
@@ -69,6 +72,71 @@ type VolumeContent struct {
 	Unpack bool `yaml:"unpack"`
 }
 
+// GadgetConnect describes an interface connection requested by the gadget
+// between seeded snaps. The syntax looks like this:
+//
+// [<plug-snap-id1>|system]:plug [[<slot-snap-id>|system]:slot]
+//
+// "system" or omitting the snap-id indicates a system plug or slot.
+// Fully omitting the slot part indicates a system slot with the same name
+// as the plug.
+type GadgetConnect struct {
+	PlugSnapID string
```

Comment:
Do we want snap names or snap IDs here? We don't use snap IDs that often. Can you add the rationale as a comment please.

Reply:
The reason we use IDs in gadgets is stability. We don't use it that often and we don't want to use it often, but if someone is cooking a device, the extra cost of listing an ID here is not so big, and it means it'll remain work no matter how names change.